### PR TITLE
Fix NetApp

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 1]
```diff
@@ -130,6 +130,8 @@ public partial class NetAppAccountResource : Azure.ResourceManager.ArmResource
...
         public virtual Azure.Pageable<Azure.ResourceManager.NetApp.Models.NetAppVault> GetVaults(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.AsyncPageable<Azure.ResourceManager.NetApp.Models.NetAppVault> GetVaultsAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Pageable<Azure.ResourceManager.NetApp.Models.NetAppVolumeGroupResult> GetVolumeGroupsByNetAppAccount(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.AsyncPageable<Azure.ResourceManager.NetApp.Models.NetAppVolumeGroupResult> GetVolumeGroupsByNetAppAccountAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:134]: Can we make this renaming? The original name reads redundant.
```suggestion
        public virtual Azure.Pageable<Azure.ResourceManager.NetApp.Models.NetAppVolumeGroupResult> GetVolumeGroups(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
        public virtual Azure.AsyncPageable<Azure.ResourceManager.NetApp.Models.NetAppVolumeGroupResult> GetVolumeGroups(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 2]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+    }
+    public partial class NetAppVolumeGroupVolume
+    {
+        public NetAppVolumeGroupVolume(string creationToken, long usageThreshold, Azure.Core.ResourceIdentifier subnetId) { }
+        public Azure.ResourceManager.NetApp.Models.NetAppAvsDataStore? AvsDataStore { get { throw null; } set { } }
+        public System.Guid? BackupId { get { throw null; } set { } }
+        public string BaremetalTenantId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier CapacityPoolResourceId { get { throw null; } set { } }
+        public int? CloneProgress { get { throw null; } }
+        public bool? CoolAccess { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1172]: Please check the name of this property in the original NetAppVolumeGroupData

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 3]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public int? CoolnessPeriod { get { throw null; } set { } }
+        public string CreationToken { get { throw null; } set { } }
+        public Azure.ResourceManager.NetApp.Models.NetAppVolumeDataProtection DataProtection { get { throw null; } set { } }
+        public long? DefaultGroupQuotaInKiBs { get { throw null; } set { } }
+        public long? DefaultUserQuotaInKiBs { get { throw null; } set { } }
+        public Azure.ResourceManager.NetApp.Models.EnableNetAppSubvolume? EnableSubvolumes { get { throw null; } set { } }
+        public bool? Encrypted { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1179]: Should be `IsEncrypted`

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 4]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public Azure.ResourceManager.NetApp.Models.NetAppEncryptionKeySource? EncryptionKeySource { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.NetApp.Models.NetAppVolumeExportPolicyRule> ExportRules { get { throw null; } }
+        public System.Guid? FileSystemId { get { throw null; } }
+        public Azure.Core.ResourceIdentifier Id { get { throw null; } }
+        public bool? IsDefaultQuotaEnabled { get { throw null; } set { } }
+        public bool? IsRestoring { get { throw null; } set { } }
+        public bool? KerberosEnabled { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1186]: IsKerberosEnabled

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 5]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public Azure.Core.ResourceIdentifier KeyVaultPrivateEndpointResourceId { get { throw null; } set { } }
+        public bool? LdapEnabled { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1188]: IsLdapEnabled

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 6]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public long? MaximumNumberOfFiles { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.NetApp.Models.NetAppVolumeMountTarget> MountTargets { get { throw null; } }
+        public string Name { get { throw null; } set { } }
+        public Azure.ResourceManager.NetApp.Models.NetAppNetworkFeature? NetworkFeatures { get { throw null; } set { } }
+        public string NetworkSiblingSetId { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1193]: Should this be arm id?

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 7]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public System.Collections.Generic.IList<Azure.ResourceManager.NetApp.Models.NetAppVolumePlacementRule> PlacementRules { get { throw null; } }
+        public System.Collections.Generic.IList<string> ProtocolTypes { get { throw null; } }
+        public string ProvisioningState { get { throw null; } }
+        public string ProximityPlacementGroup { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1197]: should this be arm-id?

**Review by ArcturusZhang — COMMENTED** [review]
Since the new class is duplicating everything in NetAppVolumeGroupData, we should make sure that the property names in them align with each other. Some of them do not align.

BTW if we could make one of them inherit from the other, it would be ideal ( but I am ok to leave them as is

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md [hunk 8]
```diff
@@ -85,14 +85,12 @@ prepend-rp-prefix:
   - ReplicationObject
   - ReplicationSchedule
   - VolumeStorageToNetworkProximity
+  - VolumeProperties
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:88]: This looks like did not have any effect on the SDK, could we remove this?

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md [hunk 9]
```diff
@@ -196,22 +197,15 @@ rename-mapping:
   VolumeRevert: NetAppVolumeRevertContent
   VolumeBackupProperties: NetAppVolumeBackupConfiguration
   VolumeGroupMetaData: NetAppVolumeGroupMetadata
+  VolumeGroup: NetAppVolumeGroupResult
+  VolumeGroupVolumeProperties.id: -|arm-id
+  VolumeGroupVolumeProperties.type: ResourceType|resource-type
+  VolumeGroupVolumeProperties: NetAppVolumeGroupVolume
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:203]: This name reads weird. How about NetAppVolumeGroupDetails? Do we have anything with same name?

**Review by ArcturusZhang — APPROVED** [review]

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1172]: fixed

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1186]: fixed

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1179]: fixed

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1193]: uuid, fixed

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md (hunk 8 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:88]: fixed

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md (hunk 9 above)
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:203]: rename to `NetAppVolumeProperties`

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 10]
```diff
@@ -1157,6 +1154,65 @@ public partial class NetAppVolumeGroupMetadata
...
+        public Azure.Core.ResourceType? ResourceType { get { throw null; } }
+        public Azure.ResourceManager.NetApp.Models.NetAppVolumeSecurityStyle? SecurityStyle { get { throw null; } set { } }
+        public Azure.ResourceManager.NetApp.Models.NetAppFileServiceLevel? ServiceLevel { get { throw null; } set { } }
+        public bool? SmbContinuouslyAvailable { get { throw null; } set { } }
+        public bool? SmbEncryption { get { throw null; } set { } }
+        public bool? SnapshotDirectoryVisible { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs]: fixed

**Pull request merged**